src/ConfigAggregator.php

```diff
@@ -92,6 +96,34 @@ public function getMergedConfig()
         return $this->config;
     }
 
+    /**
+     * Throw an exception if duplicate config providers are found.
+     *
+     * @param ProviderIterable $providers
+     * @return ProviderIterable
+     * @throws InvalidConfigProviderException
+     */
+    private function removeDuplicateProviders(iterable $providers): iterable
+    {
+        $uniqueProviders = [];
+
+        foreach ($providers as $provider) {
+            if (in_array($provider, $uniqueProviders)) {
+                if (is_object($provider)) {
+                    $provider = $provider::class;
+                }
+                if (! is_string($provider)) {
+                    $provider = serialize($provider);
```

Comment:
I don't think we should serialize the provider: risky operation, and big assumptions around it being serializable

Reply:
I'm not happy with it either, but I needed a solution to convert to string whatever is stored in `$provider` (knowing that it is not an object nor a string).
Any recommendation?

Reply:
This solution introduces more overhead than it tries to solve tbh. This will always have a performance hit while merging some config provider twice is only when config provider appears twice which is virtually never.

Reply:
Use `get_debug_type` instead, should be good to go.
An additional text like "A <whatever config provider type> is already registered" or sth. like that.
But I highly doubt that two callables can be checked via `in_array` anyways and thus, you might only end-up with `class-string` or instances of a compatible config provider. Though `get_debug_type` might be good to go here.

Reply:
I also do not expect instances being duplicated, but I've seen it a bunch of times where `class-string` was.